pkg/core/blockchain.go

```diff
@@ -394,6 +396,34 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+func (bc *Blockchain) InitOnRestore(p uint32) error {
+	if !bc.config.P2PStateExchangeExtensions || p < uint32(bc.config.StateSyncInterval) {
+		return errors.New("invalid state sync point")
+	}
+	var height uint32
+	if p > bc.config.MaxTraceableBlocks {
+		height = p - bc.config.MaxTraceableBlocks - 1
+	}
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	bc.p = p
+	if height > bc.blockHeight {
+		// Store dummy block with height specified to provide proper (bc).BlockHeight() to external users.
```

Comment:
But there is no `BlockHeight` really.

Reply:
Yes, however, we still use `bc.BlockHeight()` and `bc.HeaderHeight()` outside of the blockchain (i.e. from the server to synchronize blocks/headers for `p1` in `requestBlocks`, `handleMPTDataCmd` and `handleHeadersCmd` methods). And we need `bc` to return proper values.

Reply:
I just realised that we don't really need this dummy block, we can just set the proper `bc.blockHeight`, so you're right, I'll remove it.

Reply:
Fixed.